pkg/runner/builtin/runner.go

```diff
@@ -283,11 +235,20 @@ func cname(job *types.Job) string {
 
 func (r *Runner) Poll(q queue.Queue) {
 	for {
```

Comment:
we can use a channel here with `select`  and `time.After`

```
select {
    case <-time.After(time.Second * 10):
        ...
    }
```

I think this is the one place citadel falls short, because it doesn't allow us to block until something is available. I think the initial implementation, however, will work and we can continue to improve :)